app.js

```diff
@@ -0,0 +1,281 @@
+
+
+app.use(methodOverride('_method'));
+app.use(express.static('public'));
+app.use(bp.urlencoded({ extended: true}));
+app.engine('.hbs', exphbs({ defaultLayout: 'main', extname: '.hbs'}));
+app.set('view engine', '.hbs');
+let authorized = false;
+let onlineUser;
+
+app.get('/', (req, res) => {
+    if (!authorized) {
+        res.redirect('/login');
+    } else {
+        res.render('home');
+    }
+})
+
+//LOGIN FIRST
+app.get('/login', (req, res) => {
+    console.log('hu r u');
+    res.render('login');
+})
+
+//AUTHENTICATE USER
+app.post('/login', (req, res) => {
+    const info = req.body;
+    const user = DS_User.getUserByInfo(info.username, info.password);
+    console.log('info', info);
+    console.log('user', user);
+    if (user === undefined) {
+        console.log('Try Again')
+        res.render('login')
+    }
+    else {
+        authorized = true;
+        onlineUser = user.id;
+        console.log('Welcome!');
+        console.log('online', onlineUser);
+        res.redirect('/');
+    }
+})
+
+//SETTINGS
+app.get('/settings', (req, res) => {
+    if (!authorized) {
+        res.redirect('/login');
+    } else {
+        res.render('settings');
+    }
+})
+
+//UPDATE PASSWORD
+app.get('/updatePassword', (req, res) => {
+    if (!authorized) {
+        res.redirect('/login');
+    } else {
+        res.render('updatePassword');
+    }
+})
+
+app.put('/updatePassword', (req, res) => {
+    if (!authorized) {
+        res.redirect('/login');
+    } else {
+        const user = DS_User.getUserById(onlineUser);
+        const newPassword = req.body;
+        console.log('yes', user)
+        console.log('yes2', newPassword);
+        if (newPassword.password !== newPassword.confirm) {
+            console.log('They no match');
+            res.redirect('/updatePassword');
+        } else if (newPassword.password === newPassword.confirm) {
+            user.password = newPassword.password;
+            res.redirect('/settings');  
+        }
+    }
+})
+
+//LOGOUT
+app.get('/logout', (req, res) => {
+    authorized = false;
+    res.redirect('/login');
+})
+
+//PRODUCTS
+app.get('/products', (req, res) => {
+    if (!authorized) {
+        res.redirect('/login');
+    } else {
+        const products = DS_Prod.all();
+        console.log('products', products);
+        console.log('************');
+        res.render('products', { products });
+    }
+})
+
+//ADD PRODUCT
+app.get('/products/new', (req, res) => {
+    if (!authorized) {
+        res.redirect('/login');
+    } else {
+        res.render('productForm');
+    }
+})
+
+app.post('/products/new', (req, res) => {
+    if (!authorized) {
+        res.redirect('/login');
+    } else {
+        const product = req.body;
+        DS_Prod.add(product);
+        res.redirect('/products')
+    }
+})
+
+//PRODUCT DETAIL
+app.get('/products/:id', (req, res) => {
+    if (!authorized) {
+        res.redirect('/login');
+    } else {
+        const { id } = req.params;
```

Comment:
`let`, is a signal that the variable may be reassigned, such as a counter in a loop, or a value swap in an algorithm. It also signals that the variable will be used only in the block it’s defined in, which is not always the entire containing function. https://medium.com/javascript-scene/javascript-es6-var-let-or-const-ba58b8dcde75